src/Sylius/Component/Core/Calculator/ProductVariantPriceCalculator.php

```diff
@@ -17,7 +17,7 @@
 use Sylius\Component\Core\Model\ProductVariantInterface;
 use Webmozart\Assert\Assert;
 
-final class ProductVariantPriceCalculator implements ProductVariantPriceCalculatorInterface
+final class ProductVariantPriceCalculator implements ProductVariantPricesCalculatorInterface
```

Comment:
We need to maintain both interfaces, while `ProductVariantPricesCalculatorInterface` should extend (and maybe duplicate) `ProductVariantPriceCalculatorInterface`. `ProductVariantPriceCalculatorInterface` should marked as deprecated.

Then everywhere `ProductVariantPriceCalculatorInterface` is used, we should check if we have `ProductVariantPriceCalculatorInterface` or `ProductVariantPricesCalculatorInterface` and allow use only functions that are allowed. 

Another possibility is to completely rewrite some classes, but still, we need to maintain current contracts and behaviors in old part of code. Probably, you should go with the solution proposed by @pamil here: https://github.com/Sylius/Sylius/pull/11423#issuecomment-624504678

Reply:
If `ProductVariantPricesCalculatorInterface` extends `ProductVariantPriceCalculatorInterface` and provide the `calculateOriginal`, old contracts should be maintained ? In that case why should we check usages ? 

If `ProductVariantPriceCalculator` implements `ProductVariantPricesCalculatorInterface` that extends `ProductVariantPriceCalculatorInterface`, would there be any BC breaks left ?

Reply:
Current interfaces(ProductVariantPriceCalculatorInterface) need to be left untouched. You can create a new one(ProductVariantPricesCalculatorInterface) and extend the current one. Therefore end implementation (if the previous contract is maintained, so the old interface is still preserved there) can have new functionality. 

Same, everywhere when the old, deprecated interface is used we can check if we are using the new one, and if so make new logic available. Every place like this needs a separate deprecation note. 

That's why @pamil proposition will be the easiest one probably.